### DISCUSSION TITLE
CSLA support for client side technologies such Angular using Blazor

### DISCUSSION THREAD

**Discussion by makaranddevalekar** [MarimerLLC/csla#1812] [Questions] [answered]
Hello,

We are having a legacy application written in SilverLight with CSLA for business rules. There around more than 1000 of pages in the application and we are in process of migrating them outside SilverLight. 
We are thinking of couple of options here
1. Migrate to Blazor and Reuse the Business Logic and as CSLA has support for Blazor. We have already doing POC and it seems doable with this approach.

2. Evaluate if client side technology such Angular can be used with Blazor, and reuse business logic written in CSLA.

Would like to know your opinion if Option 2 above is possible, i.e. the UI will be rendered using Angular through Blazor but can also reuse CSLA for business logic.

Thanks in advance.

Regards,
Makarand

**Comment by rockfordlhotka** [comment, marked as answer]
Generally speaking, you don't use Blazor _and_ Angular. You use Blazor _or_ Angular, because they both fill the same niche - they are both ways to build an SPA that runs in the browser.

Angular apps are written using TypeScript, HTML, and CSS. They are not written in .NET, and so they can't include any CSLA code (because CSLA requires .NET).

Blazor apps are written using C#, HTML, and CSS. They are written in .NET, and so they can include CSLA code just fine (as per my [CSLA and Blazor book](https://store.lhotka.net/using-csla-5-blazor-and-webassembly)).

In summary, your option 2 is not really possible. You can run CSLA .NET on your web server, to help create the REST services that are called from the Angular SPA, but you can't use CSLA _in your Angular app_.

**Reply by makaranddevalekar** [reply]
Thanks for your input, this helps!
